Colorpicker adjustments

VeraZab opened the pull request:
new color picker
![ex](https://user-images.githubusercontent.com/4257572/43161803-19f6ad08-8f57-11e8-8152-b966cdb23887.gif)


demo app in react-colorscales project
![screen shot 2018-07-24 at 3 34 17 pm](https://user-images.githubusercontent.com/4257572/43161789-13c4a1a6-8f57-11e8-9a5b-8ca182467f86.png)

dmt0 commented on dev/App.js:
```diff
@@ -10,8 +10,7 @@ import PlotlyEditor, {DefaultEditor, Panel} from '../src';
 import 'brace/mode/json';
-import 'brace/theme/textmate';
-import {categoryLayout, traceTypes, chartCategory} from 'lib/traceTypes';
+import 'brace/theme/textmate'; //
```
empty comment...

dmt0 commented on src/components/fields/Colorscale.js:
```diff
@@ -30,11 +30,20 @@ class Colorscale extends Component {
+
     return (
-      <Field {...this.props}>
+      <Field {...adjustedProps}>
```
Could you just do `<Field {...this.props} label={false}>`?
I think if a props is defined twice, only the last one counts, so you can overwrite with an empty.

VeraZab replied:
that's nice, didn't think of that one thanks!

VeraZab commented:
Hey @jackparmer I just updated the react-colorscales project page, to include all api changes:
https://github.com/plotly/react-colorscales

I changed the look of the colorPicker to adapt to small UIs, but I have only extended the api, so it should be ok for the projects using it.

I've put up a few screenshots above, and deployed to gh-pages:
https://plotly.github.io/react-colorscales/

Let me know if you'd like any adjustments :)

jackparmer commented:
Hey, cool! I love it! 

Only nit that caught me eye is `Cmocean` should be all lowercase (`cmocean`). That seems to be how it's presented in the literature: https://matplotlib.org/cmocean/

![image](https://user-images.githubusercontent.com/1865834/43029751-1a5a93b6-8c57-11e8-9677-01735978a8c9.png)

nicolaskruchten commented:
I'm a bit confused by the UX of the new picker here...

![image](https://user-images.githubusercontent.com/203523/43098872-6325b4e8-8e8e-11e8-9de2-359058aa7670.png)

There's no clear indication of what the active scale is, or of how to "close" the picker, or why "reset" is there... I think I like this layout but I preferred the previous setup where the active scale is always present above, and the panel opens when you're fiddling with it and it updates the activate scale, and then you can close the panel when you're done fiddling.

VeraZab commented:
![ex](https://user-images.githubusercontent.com/4257572/43160438-21c039b8-8f53-11e8-9ad8-5710ff464c4a.gif)

ok I corrected titles, they're small caps now, just for a consistent look. 
There's also the 'scale' word in the label to make what we're selecting clearer.

@nicolaskruchten , I added back the toggle that you liked and on another UX pass, as you said, we could think of a more discoverable way to make these controls open/close.

![ex](https://user-images.githubusercontent.com/4257572/43160514-5c21eb2e-8f53-11e8-9128-34c510d6d7c2.gif)

nicolaskruchten commented on src/components/widgets/ColorscalePicker.js:
```diff
@@ -1,41 +1,95 @@
+      COLOR_PICKER_CONSTANTS.COLORSCALE_DESCRIPTIONS[selectedColorscaleType];
+    const colorscaleOptions = COLOR_PICKER_CONSTANTS.COLORSCALE_TYPES.map(
+      type => ({
+        label: type + ' scale',
```
actually now that I look at it, could we make this `scales` with an `s` please? :)

nicolaskruchten commented on src/components/widgets/ColorscalePicker.js:
```diff
@@ -1,41 +1,95 @@
+    super();
+
     this.state = {
+      selectedColorscaleType: 'sequential',
```
note: in a group of traces, this default makes less sense, and should be overrideable to `categorical`

VeraZab replied:
yup, will look at how to select the initial state better in a subsequent pr..
but the structure is there to add it easily now :)

VeraZab commented:
ok @nicolaskruchten did the latest color/swatch number modifications
![screen shot 2018-07-24 at 5 50 53 pm](https://user-images.githubusercontent.com/4257572/43168199-621abc6a-8f6a-11e8-9c51-8bc347f89eec.png)

nicolaskruchten commented:
💃
